src/main/java/com/hms_networks/americas/sc/extensions/system/application/SCAppManagement.java

```diff
@@ -56,4 +61,72 @@ public static void disableAppAutoRestart() {
     Logger.LOG_DEBUG("Disabling next JVM run command.");
     RuntimeControl.configureNextRunCommand(jvmCommand);
   }
+
+  /**
+   * Checks if a valid WAN IP address has been assigned to the device.
+   *
+   * @return true if an initialized WAN IP address is found.
+   */
+  public static boolean deviceHasWanIP() {
+    final String uninitializedWanIp = "0.0.0.0";
+    return NetManager.getWanIp().equalsIgnoreCase(uninitializedWanIp);
+  }
+
+  /**
+   * Waits for a number of minutes for a WAN IP to be established if one does not already exist.
+   * After the given time has passed, returns false indicating no WAN IP is present.
+   *
+   * @param timeout the timeout value in the specified time unit
+   * @param timeUnit the time unit of the timeout value
+   * @return true if an initialized WAN IP is detected, false otherwise
+   */
+  public static boolean waitForWanIp(int timeout, SCTimeUnit timeUnit) throws InterruptedException {
+    // Convert timeout to seconds
+    long timeoutSeconds = timeUnit.toSeconds(timeout);
+
+    // Get rounded up number of wait periods in timeout (double cast to avoid integer division)
+    int maxWaitPeriods = (int) Math.ceil((double) timeoutSeconds / (double) SECONDS_IN_WAIT_PERIOD);
+
+    boolean wanConnectionEstablished = false;
+    // Ensure there is a valid WAN IP address
+    for (int i = 0; i < maxWaitPeriods; i++) {
+
+      if (deviceHasWanIP()) {
+        // If no connection is established by end of wait time
+        if (i == (maxWaitPeriods - 1)) {
+          wanConnectionEstablished = false;
+          i = maxWaitPeriods;
+        } else {
+          waitForWanHelper(SECONDS_IN_WAIT_PERIOD);
+        }
+      } else {
+        wanConnectionEstablished = true;
+        i = maxWaitPeriods;
```

Comment:
```suggestion
        break;
```
Alternatively, you could just use `break;` here to break out of the loop.